src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -540,28 +618,166 @@ public void close()
     /**
      *
      */
-    private static class Injection
+    private class Injection
+        implements RecurringRunnable
     {
         /**
          * Ctor.
          *
-         * @param maxSeqNum
-         * @param maxTs
+         * @param receiveVideoSSRC
+         */
+        Injection(long receiveVideoSSRC)
+        {
+            this.receiveVideoSSRC = receiveVideoSSRC;
+        }
+
+        /**
+         *
+         */
+        final long receiveVideoSSRC;
+
+        /**
+         * The maximum sequence number (mod 2^16) that has been sent out by
+         * this injector instance.
+         */
+        int maxSeqNum = -1;
+
+        /**
+         * The maximum RTP timstamp (mod 2^32) that has been sent out by
+         * this injector instance.
+         */
+        long maxTs = -1;
+
+        /**
+         * A boolean indicating whether or not this instance is still injecting
+         * packets.
+         */
+        boolean running = true;
+
+        /**
+         * The last time in milliseconds at which {@link #run} was invoked.
+         */
+        long lastRunMs = -1;
+
+        /**
+         *
          */
-        public Injection(int maxSeqNum, long maxTs)
+        @Override
+        public long getTimeUntilNextRun()
         {
-            this.maxSeqNum = maxSeqNum;
-            this.maxTs = maxTs;
+            if (!running)
+            {
+                return Long.MAX_VALUE;
+            }
+
+            long timeSinceLastProcess
+                = Math.max(System.currentTimeMillis() - lastRunMs, 0);
+
+            return Math.max(PERIOD_MS - timeSinceLastProcess, 0);
         }
 
         /**
          *
          */
-        final int maxSeqNum;
+        @Override
+        public void run()
+        {
+            if (!running)
+            {
+                return;
+            }
+
+            lastRunMs = System.currentTimeMillis();
+
+            MediaStream stream = dest.getStream();
+            if (stream == null)
+            {
+                return;
+            }
+
+            synchronized (this)
```

Comment:
why do we need to synchronize this?  the only variable i can see that's also accessed elsewhere is `running`, but i don't think we actually need to synchronize in this case.  we only run the risk of an outdated value, not a potentially corrupt value, and there's always a chance running is set to false right after we check it anyway, so it seems like there isn't really a need?  but maybe i'm missing something here.